src/criteria_assessment/criteria.jl

```diff
@@ -148,6 +151,47 @@ function setup_region_routes(config, auth)
         return file(mask_path; headers=COG_HEADERS)
     end
 
+    @get auth("/suitability/assess/{reg}/{rtype}") function (
+        req::Request, reg::String, rtype::String
+    )
+        # somewhere:8000/suitability/assess/region-name/reeftype?criteria_names=Depth,Slope&lb=-9.0,0.0&ub=-2.0,40
+        # 127.0.0.1:8000/suitability/assess/Cairns-Cooktown/slopes?Depth=-4.0:-2.0&Slope=0.0:40.0&Rugosity=0.0:6.0
+
+        qp = queryparams(req)
+        @debug "In region assessment route"
+        return assess_region(reg_assess_data, reg, qp, rtype, config)
+    end
+
+    @get auth("/suitability/site-suitability/{reg}/{rtype}") function (
+        req::Request, reg::String, rtype::String
+    )
+        # 127.0.0.1:8000/suitability/site-suitability/Cairns-Cooktown/slopes?Depth=-4.0:-2.0&Slope=0.0:40.0&Rugosity=0.0:6.0&SuitabilityThreshold=95&xdist=450&ydist=50
+
+        qp = queryparams(req)
+        criteria_names = [
```

Comment:
Regarding #31 - there's no file caching implemented here.

See for example this [block of code](https://github.com/open-AIMS/ReefGuideAPI.jl/blob/aa4c0921565db4a6bfff4662e5b5e4220d4c4723/src/criteria_assessment/tiles.jl#L197-L204)

That said, it really should be moved into a function as the same pattern is used in various places so if you leave it I can implement it in a later PR.

Reply:
Oh ok, so caching the files into the tiff_cache or regional_cache still needs to be done?

Reply:
Yes, caching has to be implemented wherever we want caching, we don't have anything set up to auto-magically cache data at this stage. Except maybe if you count the browser-level cache, but that's external to the API anyway...